app/components/SocketProvider.js

```diff
@@ -125,6 +125,8 @@ class SocketProvider extends React.Component<Props> {
             if (document) {
               document.deletedAt = documentDescriptor.updatedAt;
             }
+            policies.remove(documentId);
```

Comment:
This is causing the menu with missing items here: 
https://github.com/outline/outline/issues/1095#issuecomment-687852184

I guess this is why the policy wasn't deleted, it should remain but only have the `restore` valid. This is super edge case though as you have to be viewing a document as it gets deleted, going via the trash list would load the correct policy as part of that request

Reply:
This is now resolved via https://github.com/outline/outline/pull/1517/files#r484107538